chapters/CompilerVsInterpreter.tex

```diff
@@ -0,0 +1,54 @@
+This speed increase is an effect of the compiler's architecture where it compiles the entire source code rather than interpret the source \acrfull{jit}.
+Another reason being that compiled programs often is compiled to a low level language which often have a more direct way of utilising the hardware better than a virtual environment of an interpreter.
+
+Regarding errors both model can be useful, there are strong cases for both the compiler and the interpreter model.
+A case for an interpreter and a design which catch most errors at runtime would be that design would familiarize developers of \gls{gamble} which this workflow, which would make it easier to develop the language in a direction with full or partial dynamics types, since type errors for dynamics types has to be checked at runtime.
+A wholly different case for the compiler choice would be that since the language is expecting to be untilised for large calculations with a long runtime, it would be laborious for the programmer if errors occurs at runtime, since the program could have run calculations for at long period of time.
+This case speaks for a compiler with a good error checking system, which could catch most errors a compile time before execution of the program.\citep{Sebesta, Crafting_book}
+
+When deciding between these two options it is also necessary to factor in which option has the best support for accessing the \gls{gpu}.
+Based on the research done during this project, it is the groups impression that compiled languages has advantage accessing the \gls{gpu} over virtual environments of interpreted languages.
+
+Based on these considerations it has been chosen that an compiler would be the best choice for \gls{gamble} to achieve the desired goals.
+
+\section{Target language}
+Now that it is chosen \gls{gamble} should be compiled the next choice the consider is what the target language of the compiler should be.
+The target language is the language of  the compiler's output.
+
+The starting point for finding a good target language was to go back to the different languages and libraries found during the examination of existing solutions in \myref{sec:state_of_the_art}.
+The two options that stood out the most and most other solutions seemed to be based on was CUDA by NVIDIA and OpenCL by Khronos.
+Therefore it was these two options which was considered as target language for \gls{gamble}.
+Hereafter follows a short comparison of the two options and last a choice to which solution deemed most advantageous for \gls{gamble}.
+
+Firstly it need to be made clear, to use either of these solutions, the host computer needs a framework installed to compile and execute the code.
+Both frameworks are implemented in different existing programming languages like Java, C++ and C.
+We chose to go with a C implementation due the familiarity for the project group.

```

Comment:
Aswell as the target language then being of similair structure as the source code in \gls{gamble].